File: mycollab-web/src/main/java/com/mycollab/module/project/view/message/MessageListView.java
package com.mycollab.module.project.view.message;

import com.mycollab.module.project.domain.Message;
import com.mycollab.module.project.domain.criteria.MessageSearchCriteria;
import com.mycollab.vaadin.event.HasEditFormHandlers;
import com.mycollab.vaadin.mvp.PageView;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public interface MessageListView extends PageView {
    void setCriteria(MessageSearchCriteria criteria);

    HasEditFormHandlers<Message> getEditFormHandlers();
}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/project/view/message/MessageListView.kt
package com.mycollab.mobile.module.project.view.message

import com.mycollab.mobile.ui.IListView
import com.mycollab.module.project.domain.SimpleMessage
import com.mycollab.module.project.domain.criteria.MessageSearchCriteria

/**
 * @author MyCollab Ltd.
 * @since 4.4.0
 */
interface MessageListView : IListView<MessageSearchCriteria, SimpleMessage>
